addons/sms/static/src/thread/message_patch.js

```diff
@@ -1,8 +1,38 @@
+import { _t } from "@web/core/l10n/translation";
+import { user } from "@web/core/user";
 import { Message } from "@mail/core/common/message";
 
 import { patch } from "@web/core/utils/patch";
 
 patch(Message.prototype, {
+    async onClickNotification(ev) {
+        const failureType =
+            this.message.failureNotifications.length > 0 &&
+            this.message.notification_ids[0].failure_type;
```

Comment:
Why do you only look at the first `notification` in the list? Isn't it possible that there are more than one and the failure we are looking for is not the first one?

If the notification we are looking for is always the first in the list I'm interested in the part of the code that does that :)

you can go through all `failureNotifications` and search for the one with type `sms_acc`:
```suggestion
        const failureType = this.message.notification_ids.filter(
            (notification) => notification.isFailure && notification.failure_type === "sms_acc"
        );
```

Reply:
My initial thought was that if account is unregistered then all notifications regardless of their actual error type, will have failure_type as `sms_acc`.

Since I don't really know if IAP handles it that way, its better to do the filter :)

Bonus: Even if IAP responds that way now and decides to change that, we won't have to change our code.